Source/Request.swift

```diff
@@ -387,7 +398,14 @@ public class Request {
 
     /// Appends the response serialization closure to the `Request`.
     func appendResponseSerializer(_ closure: @escaping () -> Void) {
-        protectedMutableState.write { $0.responseSerializers.append(closure) }
+        protectedMutableState.write { mutableState in
+            mutableState.responseSerializers.append(closure)
+
+            if mutableState.state == .finished {
```

Comment:
Probably want to check for `.cancelled` separately so we run the serializers either way.

Reply:
Yep, for sure. I had to add a new `responseSerializerProcessingFinished` flag in mutable state to track the completion. We can't just trigger response serializer processing when the request state is cancelled, because the regular request process will eventually get around to calling process once the queues catch up with the state change.

Reply:
Resolved in 656f331d.